### PR TITLE
core/remote: Catch errors during file uploads

### PR DESCRIPTION
**Pull request by taratatach**
When a chunk encoded request, sent via HTTP/2 fails and the remote
Cozy returns an error (e.g. 413 because the file is too large),
Chromium will replace the response header status with a simple
`net:ERR_HTTP2_PROTOCOL_ERROR`, leaving us with no information about
the reason why the request failed.
See bugs.chromium.org/p/chromium/issues/detail?id=1033945

Besides, in this situation, Electron will reject a promise with a
`mojo result is not ok` error message in a way that does not let us
catch it.
See https://github.com/electron/electron/blob/1719f073c1c97c5b421194f9bf710509f4d464d5/shell/browser/api/electron_api_url_loader.cc#L190.

Our only recourse here is to make use of the Node's
`unhandledRejection` event handler to reject our own request and
make sure the client does not hang forever on this failed request.
To make sense of the situation, we also run checks in the event
handler on the remote Cozy to try and recreate the error that was
returned by the remote Cozy and take appropriate action down the
Sync process.

Please make sure the following boxes are checked:

- [x] PR is not too big
- [x] it improves UX & DX in some way
- [x] it includes unit tests matching the implementation changes
- [x] it includes scenarios matching a new behaviour or has been manually tested
- [x] it includes relevant documentation

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -89,11 +90,12 @@
     "btoa": "^1.1.2",
     "bunyan": "^2.0.2",
     "chokidar": "^3.2.2",
-    "cozy-client": "^11.2.1",
+    "cozy-client": "^14.6.0",
     "cozy-client-js": "^0.17.1",
+    "cozy-stack-client": "^14.1.2",
```
Comment (Crash--) [inline package.json:95]: no need to add it since you import the FetchError from cozy-client

**Review by Crash-- — APPROVED** [review]
After the removal of the dep 👍

File: package.json (hunk 1 above)
Comment (taratatach) [reply package.json:95]: I need it for the linter :/

Comment (Crash--) [reply package.json:95]: OMG `const { FetchError } = require('cozy-stack-client')` my bad

**Pull request merged**